op-chain-ops/deployer/pipeline/opchain.go

```
@@ -91,9 +95,75 @@ func DeployOPChain(ctx context.Context, env *Env, bundle ArtifactsBundle, intent
+		},
+		func() {
+			setRDPImplementationAddress(ctx, env.L1Client, dco.AddressManager, &st.ImplementationsDeployment.L1CrossDomainMessengerImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, dco.L1ERC721BridgeProxy, currentBlockHash, &st.ImplementationsDeployment.L1ERC721BridgeImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, dco.L1StandardBridgeProxy, currentBlockHash, &st.ImplementationsDeployment.L1StandardBridgeImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, dco.OptimismMintableERC20FactoryProxy, currentBlockHash, &st.ImplementationsDeployment.OptimismMintableERC20FactoryImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, dco.DisputeGameFactoryProxy, currentBlockHash, &st.ImplementationsDeployment.DisputeGameFactoryImplAddress)
+		},
+	}
+
+	// Execute all set implementation address tasks concurrently
+	executeTasksConcurrently(setImplementationAddressTasks)
+
 	return nil
 }
 
+// Helper function to execute tasks concurrently
+func executeTasksConcurrently(tasks []func()) {
+	var wg sync.WaitGroup
+	for _, task := range tasks {
+		wg.Add(1)
+		go func(t func()) {
+			defer wg.Done()
+			t()
+		}(task)
+	}
+	wg.Wait()
+}
+
+func setRDPImplementationAddress(ctx context.Context, client *ethclient.Client, addressManager common.Address, implAddress *common.Address) {
+	if *implAddress == (common.Address{}) {
+		contract := opcm.NewContract(addressManager, client)
+		address, _ := contract.GetAddressByName(ctx, "OVM_L1CrossDomainMessenger")
+		*implAddress = address
+	}
+}
+
+func setEIP1967ImplementationAddress(ctx context.Context, client *ethclient.Client, proxy common.Address, currentBlockHash common.Hash, implAddress *common.Address) {
+	if *implAddress == (common.Address{}) {
+		storageValue, _ := client.StorageAtHash(ctx, proxy, genesis.ImplementationSlot, currentBlockHash)
```

This will eat the error and cause a silent failure.